src/notifications.rs

```diff
@@ -0,0 +1,32 @@
+use color_eyre::eyre::Context;
+use color_eyre::Result;
+use std::os::fd::OwnedFd;
+use tokio::net::UnixStream;
+
+pub fn create_socket() -> Result<(OwnedFd, OwnedFd)> {
+	// Create a new pair of unnamed Unix sockets
+	let (sock_1, sock_2) = UnixStream::pair().wrap_err("failed to create socket pair")?;
```

Comment:
If both sides of the pair are just going to be immediately converted with `.std_stream`, wouldn't it bit a bit more concise to just use `std::os::unix::net::UnixStream` here instead of `tokio`?